Userland/Libraries/LibWeb/HTML/HTMLIFrameElement.cpp

```diff
@@ -76,6 +76,12 @@ void HTMLIFrameElement::process_the_iframe_attributes(bool initial_insertion)
     if (!content_navigable())
         return;
 
+    // Make sure applying of history step caused by potential sync navigation to "about:blank"
+    // is finished. Otherwise, it might interrupt navigation caused by changing src or srcdoc.
+    if (!initial_insertion) {
+        main_thread_event_loop().spin_until([this] { return content_navigable_initialized(); });
+    }
```

Comment:
Same issue here. What if this causes unrelated JS execution to happen? For example if we are coming here via `someIframe.src = 'xxx'`